### PR TITLE
Python 2/3 support

### PR DESCRIPTION
**Pull request by nikolaik**
This is an attempted improvement over PR #117 which dropped python 2 compatiblity. Most tests (all except numpy tests) should pass on python 2.7 and 3.4.

Also removes some unused imports.

### CODE REVIEW

**Comment by sihrc** [comment]
Awesome, thanks! We'll take a look at it very soon.

**Comment by madisonmay** [comment]
Great stuff! I added a few comments to the diff, but planning on going ahead and merging this PR in soon + adding you to the authors list in setup.py.

File: indicoio/utils/api.py [hunk 1]
```diff
@@ -13,6 +13,10 @@ def api_handler(arg, cloud, api, url_params=None, **kwargs):
     Sends finalized request data to ML server and receives response.
     """
 
+    if type(arg) == bytes:
+        arg = arg.decode('utf-8')
```
Comment (madisonmay) [inline indicoio/utils/api.py]: Would you mind explaining the motivation for this change briefly?  Just want to fully understand how python3 is working here.

Comment (madisonmay) [inline indicoio/utils/api.py]: Nevermind, did a bit of digging into this one and the motivation for the change is clear to me now.  Seems like python3's model for dealing with unicode is vastly more sensible :)

**Comment by madisonmay** [comment]
Actually, going to make one quick change to this PR.  Going to open a PR onto our development branch instead so we can tick version numbers and the like before pushing to PyPI.  Will let you know when things are live :)

**Pull request closed without merging**